views/js/layout/search.js

```diff
@@ -13,120 +13,127 @@
  * along with this program; if not, write to the Free Software
  * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
  *
- * Copyright (c) 2014 (update and modification) Open Assessment Technologies SA;
+ * Copyright (c) 2014-2020 (update and modification) Open Assessment Technologies SA;
  */
 
 /**
  * @author Bertrand Chevrier <bertrand@taotesting.com>
  */
-define([
-    'jquery',
-    'lodash',
-    'i18n',
-    'layout/section',
-    'layout/actions',
-    'ui/feedback',
-    'ui/datatable'
-],
-function($, _, __, section, actionManager, feedback){
-    'use strict';
+define(['jquery', 'layout/actions', 'ui/searchModal', 'core/store', 'context'], function (
+    $,
+    actionManager,
+    searchModal,
+    store,
+    context
+) {
+    /**
+     * Seach bar component for TAO action bar. It exposes
+     * the container, the indexeddb store that manages
+     * search results, and @init function
+     */
+    const searchComponent = {
+        container: null,
+        searchStore: null,
+        init() {
+            store('search')
+                .then(store => {
+                    searchComponent.searchStore = store;
+                    initializeEvents();
+                    manageSearchStoreUpdate();
+                })
+                .catch(e => {
+                    actionManager.trigger('error', e);
+                });
+        }
+    };
 
     /**
-     * Create the table that contains the search results
-     * @param {Object} data - the datatable parameters
-     * @param {Object} data.model - the datatable model
-     * @param {Object} data.url
-     * @param {Object} data.params - extra parameters to give to the datatable endpoint
-     * @param {Object} data.filters - extra parameters to give to the datatable endpoint
+     * Sets events to init searchModal instance on search and results icons click, and enter keypress
      */
-    var buildResponseTable  = function buildResponseTable(data){
+    function initializeEvents() {
+        searchComponent.container = $('.action-bar .search-area');
+        const $searchBtn = $('button.icon-find', searchComponent.container);
+        const $searchInput = $('input', searchComponent.container);
+        const $resultsBtn = $('button.icon-ul', searchComponent.container);
 
-        //update the section container
-        var $tableContainer = $('<div class="flex-container-full"></div>');
-        section.updateContentBlock($tableContainer);
+        $searchBtn.off('.searchComponent').on('click.searchComponent', () => createSearchModalInstance());
```

Comment:
It is not readable when in one line

Reply:
Why do we call `off` method?
```suggestion
        $searchBtn.off('click.searchComponent').on('click.searchComponent', createSearchModalInstance);
```

Reply:
Yes, I had the same question as it was in the code I refactored. I assumed it is required to ensure that nothing else is attached to the given event. As don't fully know what other interactions could be involded there, decided to keep it in the code.